agent/stats/engine_integ_test.go

```diff
@@ -220,89 +191,275 @@ func TestStatsEngineWithNewContainers(t *testing.T) {
-	if err != nil {
-		t.Errorf("Error validating metadata: %v", err)
-	}
+	assert.NoError(t, validateMetricsMetadata(metadata), "validating metadata failed")
+	assert.Len(t, taskMetrics, 1, "incorrect number of tasks")
+	taskMetric := taskMetrics[0]
+	assert.Equal(t, aws.StringValue(taskMetric.TaskDefinitionFamily), taskDefinitionFamily, "unexcpected task definition family")
+	assert.Equal(t, aws.StringValue(taskMetric.TaskDefinitionVersion), taskDefinitionVersion, "unexcpected task definition version")
+
+	assert.NoError(t, validateContainerMetrics(taskMetric.ContainerMetrics, 1), "validating container metrics failed")
+
+	err = client.StopContainer(container.ID, defaultDockerTimeoutSeconds)
+	assert.NoError(t, err, "stopping container failed")
+	// Write the container change event to event stream
+	err = engine.containerChangeEventStream.WriteToEventStream(ecsengine.DockerContainerChangeEvent{
+		Status: api.ContainerStopped,
+		DockerContainerMetadata: ecsengine.DockerContainerMetadata{
+			DockerID: container.ID,
+		},
+	})
+	assert.NoError(t, err, "failed to write to container change event stream")
+
+	time.Sleep(waitForCleanupSleep)
+
+	// Should not contain any metrics after cleanup.
+	assert.NoError(t, validateIdleContainerMetrics(engine), "validating idle metrics failed")
+}
+
+func TestStatsEngineWithExistingContainers(t *testing.T) {
+	// Create a new docker stats engine
+	engine := NewDockerStatsEngine(&cfg, dockerClient, eventStream("TestStatsEngineWithExistingContainers"))
+
+	// Create a container to get the container id.
+	container, err := createHealthContainer(client)
+	require.NoError(t, err, "creating container failed")
+	defer client.RemoveContainer(docker.RemoveContainerOptions{
+		ID:    container.ID,
+		Force: true,
+	})
 
-	if len(taskMetrics) != 1 {
-		t.Fatalf("Incorrect number of tasks. Expected: 1, got: %d", len(taskMetrics))
+	// Wait for containers from previous tests to transition states.
+	time.Sleep(checkPointSleep)
+
+	engine.cluster = defaultCluster
+	engine.containerInstanceArn = defaultContainerInstance
+
+	err = client.StartContainer(container.ID, nil)
+	require.NoError(t, err, "starting container failed")
+	defer client.StopContainer(container.ID, defaultDockerTimeoutSeconds)
+
+	containerChangeEventStream := eventStream("TestStatsEngineWithExistingContainers")
+	taskEngine := ecsengine.NewTaskEngine(&config.Config{}, nil, nil, containerChangeEventStream, nil, dockerstate.NewTaskEngineState(), nil)
+	containers := []*api.Container{
+		{
+			Name:            "container-health",
+			HealthCheckType: "docker",
+		},
+	}
+	testTask := &api.Task{
+		Arn:                 "container-health-task",
+		DesiredStatusUnsafe: api.TaskRunning,
+		KnownStatusUnsafe:   api.TaskRunning,
+		Family:              "docker-container-health",
+		Version:             "1",
+		Containers:          containers,
 	}
+	// Populate Tasks and Container map in the engine.
+	dockerTaskEngine, _ := taskEngine.(*ecsengine.DockerTaskEngine)
```

Comment:
if you're ignoring the `bool` returned here, you can switch to the single value return version of type-casting. `dockerTaskEngine := taskEngine.(*ecsengine.DockerTaskEngine)` should work i think